cmake/SpectreTargetHeaders.cmake

```diff
@@ -0,0 +1,90 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+# Add header files for CMake to track for a given library
+# or executable.
+#
+# The header and source files together are used to track dependencies between
+# targets automatically in SpECTRE. Specifically, CI checks that the
+# dependencies of a target supply all the header files used in a library.
+# While this checking still requires includes to be correct (be they checked
+# manually or automatically), it significantly lowers the burden of correct
+# dependency management on developers.
+#
+# Additionally, some IDEs perform better if the header files for libraries
+# are known.
+#
+# Usage:
+#  spectre_target_headers(
+#    MyLib
+#    HEADERS
```

Comment:
Should this also have line with INCLUDE_DIRECTORY? Right now this usage example is inconsistent with the usage in the 3rd commit...